scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -334,7 +334,8 @@ object ScioExecutionContext {
       override def waitUntilFinish(duration: Duration, cancelJob: Boolean): ScioResult = {
         try {
           val wait = duration match {
-            case Duration.Inf => 0
+            // according to PipelineResult values <= 1 ms mean `Duration.Inf`
```

Comment:
Isn't `0 <= 1`?
> The time to wait for the pipeline to finish. Provide a value less than 1 ms for an infinite wait.

Spark runner uses `Long.MAX_VALUE` and is backed by a Java `Future.get(long, TimeUnit)`.
https://github.com/apache/beam/blob/master/runners/spark/src/main/java/org/apache/beam/runners/spark/SparkPipelineResult.java#L86
https://github.com/apache/beam/blob/master/runners/spark/src/main/java/org/apache/beam/runners/spark/SparkPipelineResult.java#L136

Reply:
This is just to be consistent with `DataflowPipelineJob` which uses the same [default value](https://github.com/apache/beam/blob/master/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java#L176).

Regarding `SparkRunner` yes that is the issue for #2085 which I believe it's a bug in `Beam`. Already made a [PR](https://github.com/apache/beam/pull/9170) for it.